perma_web/tasks/dev.py

```diff
@@ -701,3 +666,491 @@ def populate_folder_cached_path(ctx, batch_size=500):
+    merge_accounts(to_keep, to_delete)
+
+
+def merge_users_with_only_one_confirmed_account(user_list):
+    """
+    Sync all the registrars/orgs to the confirmed one and delete the other ones.
+    """
+    user_list.sort(key=lambda u: u.is_confirmed, reverse=True)
+    to_keep, *to_delete = user_list
+    merge_accounts(to_keep, to_delete)
+
+
+def merge_users_with_multiple_confirmed_accounts_but_no_links(user_list):
+    """
+    Select the account they have logged into most recently, or if they
+    have never logged in, the most recently created confirmed account.
+    Then sync all the registrars/orgs to it and delete the other ones.
+    """
+    to_keep = None
+    to_delete = []
+
+    if any(u.last_login for u in user_list):
+        for user in user_list:
+            if user.last_login:
+                if to_keep:
+                    if to_keep.last_login < user.last_login:
+                        to_delete.append(to_keep)
+                        to_keep = user
+                    else:
+                        to_delete.append(user)
+                else:
+                    to_keep = user
+            else:
+                to_delete.append(user)
+    else:
+        for user in user_list:
+            if user.is_confirmed:
+                if to_keep:
+                    if to_keep.id < user.id:
+                        to_delete.append(to_keep)
+                        to_keep = user
+                    else:
+                        to_delete.append(user)
+                else:
+                    to_keep = user
+            else:
+                to_delete.append(user)
+
+    merge_accounts(to_keep, to_delete)
+
+
+def merge_users_with_only_one_account_with_links(user_list):
+    """
+    If the account with the links is the one they have logged into most recently, keep that one:
+    sync registrars/orgs and then delete the other ones.
+
+    If the account with the links is not the one they have logged into most recently,
+    move the links to the most recently logged into account, sync the registrars/orgs,
+    and then delete the others.
+    """
+    [account_with_links] = filter(lambda u: u.link_count, user_list)
+    most_recently_logged_into_account = account_with_links
+    for user in user_list:
+        if user.last_login and user.last_login > most_recently_logged_into_account.last_login:
+            most_recently_logged_into_account = user
+
+    if account_with_links == most_recently_logged_into_account:
+        to_keep = account_with_links
+        to_delete = list(filter(lambda u: u is not to_keep, user_list))
+        merge_accounts(to_keep, to_delete)
+    else:
+        to_keep = most_recently_logged_into_account
+        to_delete = list(filter(lambda u: u is not to_keep, user_list))
+        merge_accounts(to_keep, to_delete, transfer_links=True)
+
+
+def merge_users_with_multiple_accounts_with_links(user_list):
+    """
+    We should move all links into the account with the most recent login,
```

Comment:
Not only we should, but we're doing just that 😄

Reply:
LOL whoops.